### PR TITLE
Enable stagefright and recorder for debug

### PR DESCRIPTION
**Pull request by haochuan1982**
Enable stagefright and recorder for debug

Tracked-On: OAM-111132

### CODE REVIEW

**Comment by haochuan1982** [comment]
@balajim001  @gkdeepa  @padmashree9110  @dyang23  @xinxin-wan @libinyang

**Comment by xinxin-wan** [comment]
LGTM

**Review by gkdeepa — APPROVED** [review]
LGTM

File: aosp_diff/celadon_ivi/frameworks/av/cmds/stagefright/0002-Enable-stagefright-and-recorder-for-debug.patch [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+ 
+     MediaBufferGroup *mGroup;
+diff --git a/cmds/stagefright/WavRecorder.cpp b/cmds/stagefright/WavRecorder.cpp
+new file mode 100644
+index 0000000000..7ab3b5ab9a
+--- /dev/null
++++ b/cmds/stagefright/WavRecorder.cpp
+@@ -0,0 +1,219 @@
++/*
++ * Copyright (C) 2014 The Android Open Source Project
```
Comment (JeevakaPrabu) [inline aosp_diff/celadon_ivi/frameworks/av/cmds/stagefright/0002-Enable-stagefright-and-recorder-for-debug.patch:221]: If this file is written by us, please change the copyright to Intel

File: aosp_diff/celadon_ivi/frameworks/av/cmds/stagefright/0002-Enable-stagefright-and-recorder-for-debug.patch [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+
+diff --git a/cmds/stagefright/Android.mk b/cmds/stagefright/Android.mk
+index 803c4a477a..857a0808e4 100644
+--- a/cmds/stagefright/Android.mk
++++ b/cmds/stagefright/Android.mk
+@@ -274,3 +274,30 @@ LOCAL_LICENSE_CONDITIONS:= notice
+ LOCAL_NOTICE_FILE:= $(LOCAL_PATH)/NOTICE
+ 
+ include $(BUILD_EXECUTABLE)
++#################################################################################
```
Comment (JeevakaPrabu) [inline aosp_diff/celadon_ivi/frameworks/av/cmds/stagefright/0002-Enable-stagefright-and-recorder-for-debug.patch:28]: As this introduces new unit test code, lets create new project and maintain it there. any changes to existing aosp code is ok to maintain as diff but introduction of new library/binary needs new project

**Review by JeevakaPrabu — CHANGES_REQUESTED** [review]
also add detailed commit message on the changes done in stagefright

File: aosp_diff/celadon_ivi/frameworks/av/cmds/stagefright/0002-Enable-stagefright-and-recorder-for-debug.patch (hunk 1 above)
Comment (haochuan1982) [reply aosp_diff/celadon_ivi/frameworks/av/cmds/stagefright/0002-Enable-stagefright-and-recorder-for-debug.patch]: updated

File: aosp_diff/celadon_ivi/frameworks/av/cmds/stagefright/0002-Enable-stagefright-and-recorder-for-debug.patch (hunk 2 above)
Comment (haochuan1982) [reply aosp_diff/celadon_ivi/frameworks/av/cmds/stagefright/0002-Enable-stagefright-and-recorder-for-debug.patch]: update

**Review by gkdeepa — APPROVED** [review]
LGTM

**Review by JeevakaPrabu — APPROVED** [review]

**Pull request merged**